src/pat/inject/inject.js

```diff
@@ -818,21 +818,34 @@ define([
                     if ($el.data("patterns.autoload")) {
                         return false;
                     }
+                    if (!$el.is(':visible')) {
+                        return false;
+                    }
                     if (!utils.elementInViewport($el[0])) {
                         return false;
                     }
                     // check if the target element still exists. Otherwise halt and catch fire
-                    var target = $el.data("pat-inject")[0].target.replace(/::element/, '');
-                    if ($(target).length === 0) {
-                        return false;                        
+                    var target = ($el.data("pat-inject")[0].target || cfgs[0].selector).replace(/::element/, '');
```

Comment:
@pilz This fixes the missing target definition in data-pat-inject, still I am not sure if this is complete because we can have multiple injection targets, right?